tailormap-components/projects/core/src/lib/user-interface/attributelist/attributelist-filter-values-form/attributelist-filter-values-form.component.ts

```diff
@@ -1,31 +1,68 @@
 import {
   Component,
   Inject,
+  Input,
   OnInit,
 } from '@angular/core';
 import {
   MAT_DIALOG_DATA,
   MatDialogRef,
 } from '@angular/material/dialog';
-import { FilterValueSettings } from '../attributelist-common/attributelist-filter-models';
+import { AttributeTypeEnum } from '../../../application/models/attribute-type.enum';
+import {
+  FilterValueSettings,
+  FilterDialogSettings,
+} from '../attributelist-common/attributelist-filter-models';
+import { CriteriaConditionModel } from '../../../analysis/models/criteria-condition.model';
+import { FormControl } from '@angular/forms';
 
 @Component({
   selector: 'tailormap-attributelist-filter-values-form',
   templateUrl: './attributelist-filter-values-form.component.html',
   styleUrls: ['./attributelist-filter-values-form.component.css'],
 })
+
 export class AttributelistFilterValuesFormComponent implements OnInit {
 
+  @Input()
+  public formControl: FormControl;
+
   public values: FilterValueSettings[];
+  public criteriaValue = new FormControl();
   public colName: string;
-
+  public attributeType: AttributeTypeEnum;
   public allOn: boolean
 
+  public criteria: CriteriaConditionModel;
+  public filterDialogSettings: FilterDialogSettings;
+  public filterTypeSelected: string;
+  public filterTypes: string[] = ['Contains', 'NotContains', 'UniqueValues'];
+
   constructor(private dialogRef: MatDialogRef<AttributelistFilterValuesFormComponent>,
               @Inject(MAT_DIALOG_DATA) public data: any) {
     this.colName = data.colName;
+    this.attributeType = data.attributeType;
     this.values = data.values;
-    this.allOn = true
+    this.allOn = true;
+    if (data.criteria !== null) {
+      this.criteria = data.criteria;
+      this.criteriaValue.setValue(this.criteria.value);
+    }
+    this.filterTypeSelected = data.filterType;
+    this.criteria = new class implements CriteriaConditionModel {
```

Comment:
Why new class here? Why not just
```
this.criteria = {
...
}
```
The property is already defined as `public criteria: CriteriaConditionModel;` so no need to create a class here I think